### PR TITLE
Use generic Sets at scheduler apis

### PR DESCRIPTION
**Pull request by yoongon**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Replace sets.String with sets.Set.
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Scheduler still depends on sets.String. I've just changed simple one first. However, If it is ok, I will make other module onboarded together.

#### Does this PR introduce a user-facing change?
None

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go [hunk 1]
```diff
@@ -1676,8 +1676,8 @@ func TestPreempt(t *testing.T) {
 				podInformer.GetStore().Add(test.pods[i])
 			}
 
-			deletedPodNames := make(sets.String)
-			patchedPodNames := make(sets.String)
+			deletedPodNames := sets.New[string]()
```
Comment (denkensk) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1679]: It's better to replace all `sets.String{}` in the scheduling part together in one PR.

Comment (yoongon) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1679]: Sure! I will replace all together.

Comment (chendave) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:1679]: +1 for make it in one PR for each components.

Comment (yoongon) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go]: @denkensk @chendave 
I updated scheduler apis first. Please take a look again.

Comment (kerthcet) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go]: We already have one PR here https://github.com/kubernetes/kubernetes/pull/113805, you can work on the exist one with the agreement of @Bryce-Soghigian if he doesn't have time.

Comment (yoongon) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go]: Sure I will close this PR

**Pull request closed without merging**